O2/ML/compute_efficiency_df.py

```diff
@@ -0,0 +1,193 @@
+"""
+Script for BDT efficiency calculation
+"""
+
+import argparse
+import numpy as np
+#import itertools
+import pandas as pd
+import yaml
+import matplotlib.pyplot as plt
+
+def config_bins(pt_bins):
+    """
+    Helper method to configure the bins to plot efficiency vs pT
+
+    --------------------------------
+    Parameters
+    - pt_bins: array containing the pT values used for binning
+    --------------------------------
+    Outputs
+    - bins_width: array with the width of each bin
+    - bins_mean: array with the mean of each bin
+    """
+    pt_mins = pt_bins[:-1]
+    pt_maxs = pt_bins[1:]
+    bins_width = []
+    bins_mean = []
+    for ipt, (pt_min, pt_max) in enumerate(zip(pt_mins, pt_maxs)):
+        bins_width.append(0.5*(pt_max - pt_min))
+        bins_mean.append(0.5*(pt_min + pt_max))
+    return bins_width, bins_mean
+
+def find_cut_indices(thresholds, BDT_cuts):
+    """
+    Helper method to find the index corresponding to chosen BDT cut in the 3D array eff
+
+    --------------------------------
+    Parameters
+    - thresholds: array of BDT cuts to browse
+    - BDT_cuts: array of chosen BDT output values for each class
+    --------------------------------
+    Outputs
+    - effifiency: 3D array efficiency[pT index][class][efficiency value]
+    """
+    idx_list = []
```

Comment:
This can hence be a dictionary
```suggestion
    idx_list = {}
```